pkg/plugins/golang/go_version.go

```diff
@@ -54,24 +144,13 @@ func fetchAndCheckGoVersion() error {
 // checkGoVersion should only ever check if the Go version >= 1.13, since the kubebuilder binary only cares
 // that the go binary supports go modules which were stabilized in that version (i.e. in go 1.13) by default
 func checkGoVersion(verStr string) error {
-	goVerRegex := `^go?([0-9]+)\.([0-9]+)[\.0-9A-Za-z\-]*$`
-	m := regexp.MustCompile(goVerRegex).FindStringSubmatch(verStr)
-	if m == nil {
-		return fmt.Errorf("invalid version string")
-	}
-
-	major, err := strconv.Atoi(m[1])
-	if err != nil {
-		return fmt.Errorf("error parsing major version '%s': %s", m[1], err)
-	}
-
-	minor, err := strconv.Atoi(m[2])
-	if err != nil {
-		return fmt.Errorf("error parsing minor version '%s': %s", m[2], err)
+	var version goVersion
+	if err := version.parse(verStr); err != nil {
+		return err
 	}
 
-	if major < 1 || minor < 13 {
-		return fmt.Errorf("requires version >= 1.13")
+	if version.compare(go115) < 0 || version.compare(go116alpha1) >= 0 {
```

Comment:
We cannot change here. If the person is using go 1.13 it still working.

Reply:
I'm pretty sure that go1.13 doesn't work. We didnt update the requirement here but I'm pretty sure it breaks.

Reply:
I will go back to 1.13 and track this in a separate PR.